### PR TITLE
CUMULUS-2411 - Fix lambda_invocation data resources and thin_egress_app example terraform 

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-2411](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2411)

## Changes

* Update data resource lambda invocations to run at apply time
* Update TEA cache lambda with appropriate dependencies
* Update TEA module to use appropriate count values for TF 13

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [x] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

**Comment by Jkovarik** [comment]
This is a main branch port of https://github.com/nasa/cumulus/pull/2082

File: tf-modules/cumulus/variables.tf [hunk 1]
```diff
@@ -37,6 +37,11 @@ variable "cumulus_message_adapter_lambda_layer_version_arn" {
   default     = null
 }
 
+variable "tea_map_etag" {
```
Comment (markdboyd) [inline tf-modules/cumulus/variables.tf:40]: Do we need this?

File: tf-modules/distribution/main.tf [hunk 2]
```diff
@@ -24,12 +24,13 @@ module "tea_map_cache" {
...
-  input = jsonencode({
-    bucketList = local.tea_buckets,
-    s3Bucket = var.system_bucket
-    s3Key    = "${var.prefix}/distribution_bucket_map.json"
+  function_name         = module.tea_map_cache.lambda_function_name
+  input                 = jsonencode({
+    bucketList          = local.tea_buckets,
+    s3Bucket            = var.system_bucket
+    s3Key               = "${var.prefix}/distribution_bucket_map.json"
+    teaEndpoint         =  var.tea_internal_api_endpoint
```
Comment (markdboyd) [inline tf-modules/distribution/main.tf:32]: Is this input necessary? I don't see a `teaEndpoint` property used by the tea-map-cache lambda

Comment (Jkovarik) [reply tf-modules/distribution/main.tf]: No, it's a leftover from attempting to rotate the endpoint.   
  Good catch. 3871f395fef6bfca61e64823533c3b13c3748228

**Review by markdboyd — APPROVED** [review]

**Pull request merged**